File: nim/src/conditionals.nim
proc demonstrateConditionals*() =
  echo "=== Conditionals ==="
  
  # Simple if statement
  let number = 10
  echo "number = ", number
  if number > 0:
    echo "The number is positive"
  
  # if-elif-else statement
  let x = 5
  echo "\nx = ", x
  if x > 10:
    echo "x is greater than 10"
  elif x > 0:
    echo "x is positive but not greater than 10"
  else:
    echo "x is not positive"
  
  # Nested if statements
  let
    a = 15
    b = 10
  echo "\na = ", a, ", b = ", b
  if a > 10:
    if b > 5:
      echo "Both a and b are above their thresholds"
    else:
      echo "Only a is above its threshold"
  
  # Case statement
  let fruit = "apple"
  echo "\nfruit = ", fruit
  case fruit
  of "apple":
    echo "This is an apple"
  of "banana":
    echo "This is a banana"
  of "orange":
    echo "This is an orange"
  else:
    echo "Unknown fruit"
  
  # Case statement with multiple values
  let dayNum = 2
  echo "\ndayNum = ", dayNum
  case dayNum
  of 1, 7:
    echo "Weekend"
  of 2..6:
    echo "Weekday"
  else:
    echo "Invalid day"
  
  # When statement (compile-time conditional)
  const debugging = true
  when debugging:
    echo "\nDebugging is enabled"
  
  # Using and/or in conditions
  let
    age = 25
    hasID = true
  echo "\nage = ", age, ", hasID = ", hasID
  if age >= 18 and hasID:
    echo "Can purchase alcohol"
  
  # Pattern matching with case
  type
    Shape = object
      case kind: bool
      of true:
        radius: float
      of false:
        width, height: float
  
  let
    circle = Shape(kind: true, radius: 5.0)
    rectangle = Shape(kind: false, width: 10.0, height: 20.0)
  
  echo "\nShape pattern matching:"
  case circle.kind
  of true:
    echo "Circle with radius ", circle.radius
  of false:
    echo "Rectangle"
  
  case rectangle.kind
  of true:
    echo "Circle"
  of false:
    echo "Rectangle with dimensions ", rectangle.width, " x ", rectangle.height
  
  # Using options with conditionals
  let maybeValue: Option[int] = some(42)
  echo "\nOption handling:"
  if maybeValue.isSome:
    echo "Value is present: ", maybeValue.get
  else:
    echo "No value present"
  
  echo "" # Empty line

File: r/src/conditionals.R
# Function to demonstrate conditionals in R
demonstrate_conditionals <- function() {
  cat("=== Conditionals ===\n")
  
  # Simple if statement
  x <- 10
  cat("x =", x, "\n")
  
  if (x > 5) {
    cat("x is greater than 5\n")
  }
  
  # if-else statement
  y <- 3
  cat("\ny =", y, "\n")
  if (y > 5) {
    cat("y is greater than 5\n")
  } else {
    cat("y is less than or equal to 5\n")
  }
  
  # if-else if-else statement
  z <- 0
  cat("\nz =", z, "\n")
  if (z > 0) {
    cat("z is positive\n")
  } else if (z < 0) {
    cat("z is negative\n")
  } else {
    cat("z is zero\n")
  }
  
  # Nested if statements
  a <- 15
  b <- 10
  cat("\na =", a, ", b =", b, "\n")
  if (a > 10) {
    if (b > 5) {
      cat("Both a and b are above their thresholds\n")
    }
  }
  
  # switch statement
  fruit <- "apple"
  cat("\nfruit =", fruit, "\n")
  result <- switch(fruit,
    "apple" = "This is an apple",
    "banana" = "This is a banana",
    "orange" = "This is an orange",
    "Unknown fruit"
  )
  cat("Switch result:", result, "\n")
  
  # ifelse() function (vectorized if)
  numbers <- -2:2
  cat("\nNumbers:", numbers, "\n")
  signs <- ifelse(numbers > 0, "positive", 
                 ifelse(numbers < 0, "negative", "zero"))
  cat("Signs:", signs, "\n")
  
  cat("\n")
}